tests/handlers/test_user_directory.py

```diff
@@ -25,12 +24,18 @@
 from synapse.types import create_requester
 
 from tests import unittest
+from tests.storage.test_user_directory import GetUserDirectoryTables
 from tests.unittest import override_config
 
 
-class UserDirectoryTestCase(unittest.HomeserverTestCase):
-    """
-    Tests the UserDirectoryHandler.
+class UserDirectoryTestCase(GetUserDirectoryTables, unittest.HomeserverTestCase):
```

Comment:
I'm not a fan of collecting utility functions by inheritance. It makes the dataflows harder to follow.

Make the methods in `GetUserDirectoryTables` async to avoid the dependency on `get_success`, put `self._get_user_directory_tables_helper = GetUserDirectoryTables(self.store)` in `prepare`, and then do `self.get_success(self._get_user_directory_tables_helper.get_users_in_public_rooms(...))`.